pkgs/applications/office/qownnotes/default.nix

```diff
@@ -1,29 +1,51 @@
-{ mkDerivation, lib, stdenv, fetchurl
+{ lib, stdenv, fetchurl
 , qmake, qttools, qtbase, qtsvg, qtdeclarative, qtxmlpatterns, qtwebsockets
 , qtx11extras, qtwayland
+, undmg, makeWrapper
 }:
 
-mkDerivation rec {
+let
   pname = "qownnotes";
+  appname = "QOwnNotes";
   version = "22.12.3";
 
+  inherit (stdenv.hostPlatform) system;
+  throwSystem = throw "Unsupported system: ${system}";
+
   src = fetchurl {
-    url = "https://download.tuxfamily.org/${pname}/src/${pname}-${version}.tar.xz";
-    # Fetch the checksum of current version with curl:
-    # curl https://download.tuxfamily.org/qownnotes/src/qownnotes-<version>.tar.xz.sha256
-    sha256 = "sha256-fpI7RYOGmWwmau6tF8FPmY2/FtN9foWRX8/WgrNU6E8=";
+    url = {
+      x86_64-linux = "https://download.tuxfamily.org/${pname}/src/${pname}-${version}.tar.xz";
+      x86_64-darwin = "https://github.com/pbek/QOwnNotes/releases/download/v${version}/${appname}.dmg";
+    }.${system} or throwSystem;
+    sha256 = {
+      x86_64-linux = "0kz8ajrq5mngby8qazkxscbbz3cr9z0igbgfd8k6r6c6hd2kp4ky";
+      x86_64-darwin = "1y9r7fbilhsynzm8527d20kv68x3s475j6vv4rss4kk12av7qfks";
+    }.${system} or throwSystem;
   };
 
-  nativeBuildInputs = [ qmake qttools ];
+  meta = with lib; {
+      description = "Plain-text file notepad and todo-list manager with markdown support and Nextcloud/ownCloud integration";
+      homepage = "https://www.qownnotes.org/";
+      downloadPage = "https://github.com/pbek/QOwnNotes/releases";
+      license = licenses.gpl2Only;
+      maintainers = with maintainers; [ totoroot ];
+      platforms = [ "x86_64-linux" "x86_64-darwin" ];
+  };
 
-  buildInputs = [ qtbase qtsvg qtdeclarative qtxmlpatterns qtwebsockets qtx11extras ]
-    ++ lib.optionals stdenv.isLinux [ qtwayland ];
+  linux = stdenv.mkDerivation {
+    inherit pname version src meta;
+    nativeBuildInputs = [ qmake qttools qtbase qtsvg qtdeclarative qtxmlpatterns qtwebsockets qtx11extras qtwayland ];
```

Comment:
I guess most of them should go to `buildInputs`.

Reply:
You're right. I took a look at some other Qt packages and fixed a few issues in 4ccfb32.
QOwnNotes was the first package I added to nixpkgs and I hadn't changed much since.